mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -132,24 +160,31 @@ where
             .await
         {
             Ok(result) => result,
-            Err(ObjectClientError::ServiceError(GetObjectError::NoSuchKey)) => return Ok(None),
-            Err(e) => return Err(e.into()),
+            Err(err) => return self.handle_get_object_err(err),
```

Comment:
it looks that `ObjectClientError::ServiceError(GetObjectError::NoSuchKey)` is not even possible here, but I'd check what this [method](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3-client/src/s3_crt_client/get_object.rs#L125) returns